File: code/binding/development/subarchitectures/planning.sa/src/java_old/planning/util/PlanningUtils.java
package planning.util;

/**
 * 
 */

import planning.autogen.PlanningProblem;
import Planner.Fact;
import Planner.ObjectDeclaration;
import Planner.PlanningTask;

/**
 * @author nah
 */
public class PlanningUtils {

	/**
	 * @param _planningProblem
	 * @return
	 */
	public static String toString(PlanningProblem _planningProblem) {
		String out = "PlanningProblem:\n";

		out += _planningProblem.m_maplGoal;
		out += "\nObjects:\n";
		for (int i = 0; i < _planningProblem.m_objectList.length; i++) {
			out += _planningProblem.m_objectList[i].name + " : "
					+ _planningProblem.m_objectList[i].type + "\n";
		}
		out += "\nFacts:\n";
		for (int i = 0; i < _planningProblem.m_factList.length; i++) {
			out += _planningProblem.m_factList[i] + "\n";
		}

		return out;
	}

	// public static String toString(PlanningState _state) {
	// String out = "PlanningState:\n";
	//
	// out += "\nObjects:\n";
	// for (int i = 0; i < _state.m_objects.length; i++) {
	// out += _state.m_objects[i].name + " : " + _state.m_objects[i].type
	// + "\n";
	// }
	// out += "\nFacts:\n";
	// for (int i = 0; i < _state.m_facts.length; i++) {
	// out += _state.m_facts[i] + "\n";
	// }
	//
	// return out;
	// }
	//
	// /**
	// * @param _plan
	// * @return
	// */
	// public static String toString(Plan _plan) {
	// String out = "Plan:\n";
	// for (int i = 0; i < _plan.m_plan.length; i++) {
	// out += i + ": " + _plan.m_plan[i] + "\n";
	// }
	// return out;
	// }

	/**
	 * @param _decl
	 * @return
	 */
	public static String toString(ObjectDeclaration _decl) {
		return "(" + _decl.name + " - " + _decl.type + ")";
	}

	/**
	 * @param _decl
	 * @return
	 */
	public static String toString(Fact _fact) {
		StringBuffer sb = new StringBuffer("(");
		sb.append(_fact.name);
		sb.append(" ");
		for (String arg : _fact.arguments) {
			sb.append(arg);
			sb.append(" ");
		}
		sb.append(": ");
		sb.append(_fact.value);
		sb.append(")");
		return sb.toString();
	}

	public static String toString(PlanningTask _task) {

		StringBuffer sb = new StringBuffer("(");
		sb.append(_task.task_id);
		sb.append("\n");
		sb.append(_task.domain_fn);
		sb.append("\n");
		sb.append(_task.domain_name);
		sb.append("\n");
		sb.append(_task.planning_agent);
		sb.append("\n");
		sb.append("GOALS:\n");
		for (String arg : _task.goals) {
			sb.append(arg);
			sb.append("\n");
		}
		sb.append("OBJECTS:\n");

		for (ObjectDeclaration arg : _task.objects) {
			sb.append(PlanningUtils.toString(arg));
			sb.append("\n");
		}
		sb.append("FACTS:\n");

		for (Fact arg : _task.facts) {
			sb.append(PlanningUtils.toString(arg));
			sb.append("\n");
		}

		return sb.toString();
	}

	// /**
	// * @param _state
	// * @return
	// */
	// public static String toString(TemporaryPlanningState _state) {
	// return toString(_state.toPlanningState());
	// }

}

File: binding/development/subarchitectures/planning.sa/src/c++/planning/util/PlanningUtils.cpp
#include "PlanningUtils.hpp"
using namespace Planner;


bool operator== (const Fact & _f1, 
		 const Fact & _f2) {

  if (strcmp(_f1.name,_f2.name) != 0 ||
      _f1.arguments.length() != _f2.arguments.length()) {
    return false;
  }

  // are the arguments the same?
  for (unsigned int i = 0; i < _f1.arguments.length(); i++) {
    if (strcmp(_f1.arguments[i],_f2.arguments[i]) != 0) {
      return false;
    }
  }

  // are the values the same?
  return (strcmp(_f1.value,_f2.value) == 0);

}








std::ostream & 
operator<<(std::ostream &_stream, 
	   const Planner::Fact &_fact) {

  if (_fact.modality == K_MODALITY) {
    _stream<<"(K ";
  }

  _stream<<"(";
  _stream<<_fact.name;
  _stream<<" ";
  for (unsigned int i = 0; i <  _fact.arguments.length(); ++i) {
    _stream<<_fact.arguments[i];
    _stream<<(" ");
  }
  _stream<<": ";
  _stream<<_fact.value;
  _stream<<")";

  if (_fact.modality == K_MODALITY) {
    _stream<<")";
  }

  return _stream;
}

std::ostream & 
operator<<(std::ostream &_stream, 
	   const Planner::ObjectDeclaration &_decl) {
  _stream<<"(" << _decl.name<< " - " << _decl.type << ")";
  return _stream;
}


std::ostream & 
operator<<(std::ostream &_stream, 
	   const TemporaryPlanningState &_state) {
  _stream<<"TemporaryPlanningState:"<<endl;

  _stream <<"\nObjects:"<<endl;
  for(TemporaryPlanningState::ObjectDeclarationSet::const_iterator os = _state.m_objectList.begin(); 
      os != _state.m_objectList.end(); ++os) {    
      _stream <<*os<<endl;
  }

  _stream <<"\nFacts:"<<endl;
  for(TemporaryPlanningState::FactSet::const_iterator fs = _state.m_factList.begin(); 
      fs != _state.m_factList.end(); ++fs) {          
      _stream <<*fs<<endl;
  }
  return _stream;
}

